virtool/blast/data.py

```diff
@@ -16,10 +16,10 @@ class BLASTData:
     A data layer domain for NuVs BLAST data.
     """
 
-    def __init__(self, db, pg: AsyncEngine, tasks: TasksClient):
+    def __init__(self, db, pg: AsyncEngine, tasks_data: TasksData):
         self._db = db
         self._pg = pg
-        self._tasks = tasks
+        self._tasks_data = tasks_data
```

Comment:
Use `self.data` like in analyses. You will have to call `bind_layer`. That is not in `__post_init__` yet.

Do this for all further situations.